lerobot/common/policies/vla/modeling_vla.py

```diff
@@ -78,299 +66,210 @@ def reset(self):
-        lora_alpha=16,  # Scaling factor
-        lora_dropout=0.1,  # Dropout applied to LoRA layers
-        target_modules=["q_proj", "v_proj"]  # The attention components where LoRA is applied
-        )
-        self.lora_config = lora_config
-        for param in self.vision_language_model.parameters():
-            param.requires_grad = False
+            # Apply LoRA and ensure only LoRA parameters are trainable
+            self.vision_language_model = get_peft_model(self.vision_language_model, lora_config)
+            for name, param in self.vision_language_model.named_parameters():
+                if "lm_head" in name:
+                    param.requires_grad = True
 
-        # Apply LoRA and ensure only LoRA parameters are trainable
-        self.vision_language_model = get_peft_model(self.vision_language_model, lora_config)
-        for name, param in self.vision_language_model.named_parameters():
-            if "lm_head" in name:  # Adjust "lm_head" to the specific name of the head layer
-                param.requires_grad = True
         # Verify trainable parameters
         trainable_params = []
         for name, param in self.vision_language_model.named_parameters():
             if param.requires_grad:
                 trainable_params.append(name)
                 print(f"Trainable parameter: {name}")
 
-        #self.device = self.vision_language_model.device
-        #self.half()
+    def apply_prompt_template(self, text: str, add_generation_prompt: bool = True) -> str:
+        if "llava-onevision" in self.vlm_backbone_name:
+            conversation = [
+                {
+                    "role": "user",
+                    "content": [
+                        {"type": "text", "text": text},
+                        {"type": "image"},
+                    ],
+                },
+            ]
+            prompt = self.processor.apply_chat_template(
+                conversation, add_generation_prompt=add_generation_prompt
+            )
+        elif "paligemma" in self.vlm_backbone_name:
+            prompt = f"<image>{text}"
+        else:
+            prompt = text
+
+        return prompt
+
+    def get_vlm_features(self, processed_inputs) -> torch.Tensor:
+        vlm_output = self.vision_language_model(
+            **processed_inputs, return_dict=True, output_hidden_states=True
+        )
 
-    def forward(self, processed_inputs, batch_size):
+        if any(k in self.vlm_backbone_name for k in ["llava-onevision", "paligemma"]):
+            batch_size = processed_inputs["input_ids"].shape[0]
+            last_hidden_state = vlm_output.hidden_states[-1]
+            seq_len = vlm_output.image_hidden_states.shape[0] // batch_size
+            image_features = vlm_output.image_hidden_states.view(batch_size, seq_len, -1)
+
+            if self.vlm_backbone_feature_selection == "first_image":
+                num_img_feats = 598  # this is specific to llava-onevision
```

Comment:
we can move this to config as well. not really specific to llava. As I mentioned this was to match the number of hidden states in ACT